clusterloader2/pkg/prometheus/experimental.go

```diff
@@ -199,3 +211,21 @@ func (pc *Controller) tryDeletePrometheusDisk(pdName, zone string) error {
 	klog.V(2).Infof("Deleting disk finished with: %q", string(output))
 	return nil
 }
+
+func (pc *Controller) EnableTearDownPrometheusStackOnInterrupt() {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	go func() {
+		for {
+			// Block until a signal is received.
+			s := <-c
+			fmt.Println("Got signal:", s)
+			go func() {
```

Comment:
Why do you do that in goroutine?

Reply:
I guess that by design I wanted to handle receiving signal as fast as possible, then delegating real work to additional thread. This way we are not blocked on time required to handle signal. Something like Top and Bottom Halves for Linux kernel.

Agree that in this case we do not really need that - after first SIGINT we will exit CL2.

Removing loop and additional goroutine.